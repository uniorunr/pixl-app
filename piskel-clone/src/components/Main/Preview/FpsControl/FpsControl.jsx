import React, { Component } from 'react';
import './FpsControl.scss';
import PropTypes from 'prop-types';

class FpsControl extends Component {
  constructor(props) {
    super(props);
    this.inputRef = React.createRef();
  }

  componentDidMount = () => {
    const { updateFps } = this.props;
    updateFps(this.inputRef.current.value);
  };

  handleInputChange = ({ target: { value } }) => {
    const { updateFps } = this.props;
    updateFps(value);
  };

  render() {
    const { fps } = this.props;

    return (
      <div className="fps-control">
        <label htmlFor="fps-control__input">
          <span className="fps-control__label">FPS: {fps}</span>
          <input
            type="range"
            id="fps-control__input"
            className="fps-control__input"
            name="volume"
            min="1"
            max="24"
            defaultValue="12"
            step="1"
            onInput={this.handleInputChange}
            ref={this.inputRef}
          />
        </label>
      </div>
    );
  }
}

FpsControl.propTypes = {
  updateFps: PropTypes.func.isRequired,
  fps: PropTypes.number.isRequired,
};

export default FpsControl;
