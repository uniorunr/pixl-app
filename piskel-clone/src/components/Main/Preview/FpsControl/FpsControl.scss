.fps-control {
  label {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;

    span:nth-child(2) {
      margin-right: 5px;
    }

    .fps-control__input {
      width: 100px;
      flex-grow: 1;
      cursor: pointer;
    }
  }
}
