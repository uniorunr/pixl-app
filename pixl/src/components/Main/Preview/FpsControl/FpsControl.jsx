import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import * as actions from '../../../../actions/actions';
import './FpsControl.scss';

class FpsControl extends Component {
  inputRef = React.createRef();

  componentDidMount = () => {
    const { updateFps } = this.props;
    updateFps(+this.inputRef.current.value);
  };

  handleInputChange = ({ target: { value } }) => {
    const { updateFps } = this.props;
    updateFps(+value);
  };

  render() {
    const { fps } = this.props;

    return (
      <div className="fps-control">
        <label htmlFor="fps-control__input">
          <span className="fps-control__label">FPS: {fps}</span>
          <input
            type="range"
            id="fps-control__input"
            className="fps-control__input"
            name="volume"
            min="1"
            max="24"
            defaultValue="12"
            step="1"
            onInput={this.handleInputChange}
            ref={this.inputRef}
          />
        </label>
      </div>
    );
  }
}

FpsControl.propTypes = {
  updateFps: PropTypes.func.isRequired,
  fps: PropTypes.number.isRequired,
};

const mapStateToProps = state => ({
  fps: state.preview.fps,
});

const mapDispatchToProps = (dispatch) => {
  const { updateFps } = bindActionCreators(actions, dispatch);
  return {
    updateFps: (fps) => {
      updateFps(fps);
    },
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(FpsControl);
