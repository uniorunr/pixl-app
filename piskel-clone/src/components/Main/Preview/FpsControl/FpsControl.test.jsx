import React from 'react';
import renderer from 'react-test-renderer';
import FpsControl from './FpsControl';

it('renders correctly', () => {
  const tree = renderer.create(<FpsControl />).toJSON();
  expect(tree).toMatchSnapshot();
});
