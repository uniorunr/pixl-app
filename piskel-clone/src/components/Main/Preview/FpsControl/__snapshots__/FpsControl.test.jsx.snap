// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="fps-control"
>
  <label
    htmlFor="fps-control__input"
  >
    <span />
    <span>
      FPS
    </span>
    <input
      className="fps-control__input"
      id="fps-control__input"
      max="24"
      min="1"
      name="volume"
      step="1"
      type="range"
    />
  </label>
</div>
`;
